internal/kafka/internal/workers/clusters_mgr_test.go

```diff
@@ -337,13 +335,13 @@ func TestClusterManager_processMetrics(t *testing.T) {
 					ClusterService:         tt.fields.clusterService,
 					DataplaneClusterConfig: tt.fields.dataplaneClusterConfig,
 					SupportedProviders:     tt.fields.supportedProviders,
-					OCMConfig:              tt.fields.OCMConfig,
-					ProviderFactory:        tt.fields.ProviderFactory,
+					OCMConfig:              tt.fields.ocmConfig,
+					ProviderFactory:        tt.fields.providerFactory,
 				},
 			}
 			// processMetrics accumulates all the errors encountered during metrics processing in an array.
-			// If that array is non empty then an error should be expected. Otherwise, no errors should be expected.
-			g.Expect(len(c.processMetrics()) > 0).To(gomega.Equal(tt.wantErr))
+			gotErrors := c.processMetrics()
+			g.Expect(len(gotErrors)).To(gomega.Equal(len(tt.wantErrs)), "Errors received: %v", gotErrors)
```

Comment:
Can we use `Equal` so that we will check that the received errors will be precisely the expected errors?

Reply:
Can't use equal on the error list itself unfortunately because the error stack will be different. What we could have here though is check the error message and ensure they're what we expect. wdyt?

Reply:
Added this in https://github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/pull/1251/commits/59c21cb7be738860e05428f734666a969fcddc0f, let me know what you think